compiler/rustc_middle/src/ty/consts/valtree.rs

```diff
@@ -63,6 +63,12 @@ impl<'tcx> ValTree<'tcx> {
         Self::Branch(interned)
     }
 
+    pub fn from_scalars<'a>(tcx: TyCtxt<'tcx>, scalars: &'a [ScalarInt]) -> Self {
```

Comment:
```suggestion
    pub fn from_scalars(tcx: TyCtxt<'tcx>, scalars: impl IntoIter<Item = ScalarInt>) -> Self {
```

then you can avoid `collect` at the call sites to `from_scalars`